k4MarlinWrapper/src/components/MarlinProcessorWrapper.cpp

```diff
@@ -155,8 +155,21 @@ StatusCode MarlinProcessorWrapper::initialize() {
     streamlog::out.init(std::cout, "k4MarlinWrapper");
     marlin::Global::parameters = new marlin::StringParameters();
     marlin::Global::parameters->add("AllowToModifyEvent", {"true"});
-    marlin::Global::parameters->add("RandomSeed", {"123456"});
-    // marlin::Global::EVENTSEEDER = new marlin::ProcessorEventSeeder() ;
+
+    // Random Service setup
+    std::vector<long> gotSeeds{};
+    StatusCode        set_seeds_sc = randSvc()->engine()->seeds(gotSeeds);
+    if (!set_seeds_sc.isSuccess())
+      warning() << "Random Service seeds could not be set" << endmsg;
+
+    auto sc_flatGen_init = m_flatGenerator.initialize(randSvc(), Rndm::Flat(0, 999999));
+    if (!sc_flatGen_init.isSuccess())
+      return Error("Cannot initialize Random Service based flat generator");
+
+    std::string marlin_seed = std::to_string((long)m_flatGenerator());
```

Comment:
As it is right now it will just fail at initialization and won't be able to even execute. If no seed is set, the default one is used (as it is right now).